gguf-py/gguf/utility.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+
+def fill_templated_filename(filename: str, output_type: str):
+    # Given a file name fill in any type templates e.g. 'some-model-name.{ftype}.gguf'
+    ftype_uppercase: str = output_type.upper()
+    ftype_lowercase: str = output_type.lower()
+    return filename.format(ftype_lowercase,
+                           outtype=ftype_lowercase, ftype=ftype_lowercase,
+                           OUTTYPE=ftype_uppercase, FTYPE=ftype_uppercase)
+
+
+def model_weight_count_rounded_notation(model_params_count: int) -> str:
+    if model_params_count > 1e15 :
+        # Quadrillion Of Parameters
+        scaled_model_params = model_params_count * 1e-15
+        scale_suffix = "Q"
+    elif model_params_count > 1e12 :
+        # Trillions Of Parameters
+        scaled_model_params = model_params_count * 1e-12
+        scale_suffix = "T"
+    elif model_params_count > 1e9 :
+        # Billions Of Parameters
+        scaled_model_params = model_params_count * 1e-9
+        scale_suffix = "B"
+    elif model_params_count > 1e6 :
+        # Millions Of Parameters
+        scaled_model_params = model_params_count * 1e-6
+        scale_suffix = "M"
+    else:
+        # Thousands Of Parameters
+        scaled_model_params = model_params_count * 1e-3
+        scale_suffix = "K"
+    return f"{round(scaled_model_params)}{scale_suffix}"
+
+
+def size_label(expert_count_int:int | None, model_params_count: int) -> str:
+    per_model_rounded_weight_estimate = model_weight_count_rounded_notation(model_params_count)
+
+    if expert_count_int is not None and expert_count_int > 0:
+        size_class = f"{expert_count_int}x{per_model_rounded_weight_estimate}"
+    else:
+        size_class = f"{per_model_rounded_weight_estimate}"
+
+    return size_class
+
+
+def naming_convention(model_name: str | None, base_name: str | None, finetune_string:str | None, version_string:str | None, size_label: str | None, output_type: str | None) -> str:
```

Comment:
Maybe `output_type` can take `gguf.LlamaFileType` as type, then inside this function we can decide how to format it to string